cap-primitives/src/yanix/linux/fs/open.rs

```diff
@@ -95,37 +119,13 @@ fn open_openat2(start: &fs::File, path: &Path, options: &OpenOptions) -> io::Res
     open_manually_wrapper(start, path, options)
 }
 
-lazy_static! {
-    static ref OPEN: fn(&fs::File, &Path, &OpenOptions) -> io::Result<fs::File> = {
-        // Test if `openat2` is supported. If so, we can use `open_openat2`.
-        // Otherwise, fall back to `open_manually`.
-        unsafe {
-            if let -1 = libc::syscall(
-                SYS_OPENAT2,
-                -1,
-                ptr::null::<libc::c_void>(),
-                ptr::null::<libc::c_void>(),
-                0,
-            ) {
-                if let libc::ENOSYS = io::Error::last_os_error().raw_os_error().unwrap() {
-                    // We're on an older Linux.
-                } else {
-                    return open_openat2;
-                }
-            }
-        }
-
-        open_manually_wrapper
-    };
-}
-
 #[inline]
 pub(crate) fn open_impl(
     start: &fs::File,
     path: &Path,
     options: &OpenOptions,
 ) -> io::Result<fs::File> {
-    OPEN(start, path, options)
+    openat2_or_open_manually(start, path, options)
```

Comment:
`open_impl` here is now just a wrapper around `openat2_or_open_manually` that doesn't do anything on its own. Would it make sense to make `openat2_or_open_manually` be `pub(crate)`, and then have the mod.rs file do `use openat2_or_open_manually as open_impl`?

Reply:
Hmm, I like the direction of this, but I think it would be even better if we simply renamed `openat2_or_open_manually` to `open_impl`. This way, we don't have to tweak any other bit which uses `fs::open_impl`. Otherwise, since `openat2_or_open_manually` already resides in `fs` module, we couldn't simply re-export it as `open_impl` in `fs` mod in `mod.rs`, since `fs` is already defined. What do you think?

Reply:
Sounds good to me!